File: file-service/src/main/java/com/saintrivers/file/config/rabbit/QueueSender.java
package com.saintrivers.file.config.rabbit;

import com.saintrivers.common.dto.upload.UploadEvent;
import lombok.RequiredArgsConstructor;
import org.springframework.amqp.core.Queue;
import org.springframework.amqp.rabbit.core.RabbitTemplate;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class QueueSender {

    private final RabbitTemplate rabbitTemplate;
    private final Queue queue;

    public void send(UploadEvent uploadEvent){
        rabbitTemplate.convertAndSend(queue.getName(), uploadEvent);
    }

}

File: storage-service/src/main/kotlin/com/saintrivers/storage/config/rabbit/QueueSender.kt
package com.saintrivers.storage.config.rabbit

import com.saintrivers.common.dto.UploadEvent
import org.springframework.amqp.core.Queue
import org.springframework.amqp.rabbit.core.RabbitTemplate
import org.springframework.stereotype.Component

@Component
class QueueSender(val rabbitTemplate: RabbitTemplate, val queue: Queue) {

    fun send(uploadEvent: UploadEvent) {
        rabbitTemplate.convertAndSend(this.queue.name, uploadEvent)
    }
}